### PR TITLE
Add support for "Grouped products" 

### PR DESCRIPTION
**Pull request by vladimirsm**
Added support for grouped products.
This solves #38

### CODE REVIEW

File: src/app/component/CartItem/CartItem.component.js [hunk 1]
```diff
@@ -37,7 +37,7 @@ class CartItem extends Component {
 
         return {
             pathname: `/product/${ url_key }`,
-            state: { product, variantIndex },
+            state: { product: product.parent || product, variantIndex },
```
Comment (alfredsgenkins) [inline src/app/component/CartItem/CartItem.component.js]: Can you destruct this?

File: src/app/component/GroupedProductsList/GroupedProductsList.component.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+/**
+ * Product description
+ * @class GroupedProductList
+ */
+class GroupedProductList extends Component {
+    renderProductList(items) {
+        return (
+            <ul>
+                { items
```
Comment (alfredsgenkins) [inline src/app/component/GroupedProductsList/GroupedProductsList.component.js]: Bad wrap

File: src/app/component/ProductActions/ProductActions.component.js [hunk 3]
```diff
@@ -87,22 +87,46 @@ class ProductActions extends Component {
     addProduct() {
         const {
             addProduct, product, product: {
-                variants
+                variants, type_id
             }, configurableVariantIndex
         } = this.props;
         const { itemCount } = this.state;
 
+        if (type_id === 'grouped') {
```
Comment (alfredsgenkins) [inline src/app/component/ProductActions/ProductActions.component.js]: Please make a case switch for configurable and grouped here.

File: src/app/component/ProductDetails/ProductDetails.component.js [hunk 4]
```diff
@@ -25,15 +25,15 @@ class ProductDetails extends Component {
      * Render product SKU only when it's loaded
      */
     renderSku() {
-        const { product: { variants }, areDetailsLoaded, configurableVariantIndex } = this.props;
+        const { product: { variants, sku }, areDetailsLoaded, configurableVariantIndex } = this.props;
 
-        if (variants && variants[configurableVariantIndex] && areDetailsLoaded) {
-            const { product } = variants[configurableVariantIndex];
+        if (areDetailsLoaded) {
+            const { product } = variants && configurableVariantIndex ? variants[configurableVariantIndex] : '';
```
Comment (alfredsgenkins) [inline src/app/component/ProductDetails/ProductDetails.component.js]: You are going to destruct `''`?

File: src/app/route/ProductPage/ProductPage.component.js [hunk 5]
```diff
@@ -196,13 +205,22 @@ class ProductPage extends Component {
...
                           configurableVariantIndex={ configurableVariantIndex }
                         />
-                        <ProductActions
-                          product={ dataSource }
-                          availableFilters={ filters }
-                          configurableVariantIndex={ configurableVariantIndex }
-                          areDetailsLoaded={ areDetailsLoaded }
-                          updateConfigurableVariantIndex={ index => this.updateUrl(index) }
-                        />
+                        <div>
```
Comment (alfredsgenkins) [inline src/app/route/ProductPage/ProductPage.component.js]: Please pay attention to accessibility. What does this div represent?

File: src/app/store/Product/Product.dispatcher.js [hunk 6]
```diff
@@ -28,7 +28,7 @@ class ProductDispatcher extends RequestDispatcher {
     onSuccess(data, dispatch) {
         const { products: { items, filters } } = data;
         const productItem = items[0];
-
+        const product = productItem.type_id === 'grouped' ? this._prepareGroupedProduct(productItem) : productItem;
```
Comment (alfredsgenkins) [inline src/app/store/Product/Product.dispatcher.js]: Please make it two line.

File: src/app/store/Product/Product.dispatcher.js [hunk 7]
```diff
@@ -28,7 +28,7 @@ class ProductDispatcher extends RequestDispatcher {
     onSuccess(data, dispatch) {
         const { products: { items, filters } } = data;
         const productItem = items[0];
```
Comment (alfredsgenkins) [inline src/app/store/Product/Product.dispatcher.js]: Please destruct productItem!

**Comment by alfredsgenkins** [comment]
Let's also make sure all the functions are commented! :) @vladimirsm

File: src/app/route/ProductPage/ProductPage.component.js (hunk 5 above)
Comment (vladimirsm) [reply src/app/route/ProductPage/ProductPage.component.js]: This div is needed to display grouped products together with add to cart button. I added aria-label to identify what this div is for.

File: src/app/component/ProductDetails/ProductDetails.component.js (hunk 4 above)
Comment (vladimirsm) [reply src/app/component/ProductDetails/ProductDetails.component.js]: this destructs variants[configurableVariantIndex], and if it doesn't exists product becomes ''

**Pull request merged**